tests/core/dom/range/extractcontents.js

```diff
@@ -379,6 +381,35 @@
 
 			// See: execContentsAction in range.js.
 			assert.isInnerHtmlMatching( '<p>Foo bar</p>', docFrag.getHtml(), 'Extracted HTML' );
+		},
+
+		// #644
+		'test extract nested tags': function() {
+			var range = new CKEDITOR.dom.range( doc ),
+				docFrag;
+
+			// You want to select text nodes.
+			range.setStart( doc.getById( '_pe' ).getFirst(), 0 );
+			range.setEnd( doc.getById( '_sub' ).getFirst(), 3 );
+
+			docFrag = range.extractContents();
+
+			assert.isInnerHtmlMatching( '111<strong>222<span>333<em>444</em></span>555<sup>666</sup>777<sub id="_sub">888</sub></strong>', docFrag.getHtml() );
+
+		},
+
+		// #644
+		'test extractContents and cloneContents provides equal results': function() {
+			var range = new CKEDITOR.dom.range( doc ),
+				extractFrag, cloneFrag;
+
+			range.setStart( doc.getById( '_pe' ).getFirst(), 0 );
+			range.setEnd( doc.getById( '_sub' ).getFirst(), 3 );
+
+			cloneFrag = range.cloneContents(),
+			extractFrag = range.extractContents();
+
+			assert.isInnerHtmlMatching( bender.tools.fixHtml( cloneFrag.getHtml() ), bender.tools.fixHtml( extractFrag.getHtml() ) );
```

Comment:
This line got long and ugly. Rather than using `assert.beautified.html` multiple times, how about assert using beautified HTML?

`assert.beautified.html( cloneFrag.getHtml(), extractFrag.getHtml() );`